x-pack/plugins/observability_solution/dataset_quality/public/state_machines/dataset_quality_controller/src/state_machine.ts

```diff
@@ -632,6 +658,13 @@ export const createPureDatasetQualityControllerStateMachine = (
             : {};
         }),
       },
+      guards: {
+        checkIfActionForbidden: (context, event) => {
+          return (
+            'data' in event && 'message' in event.data && /forbidden/i.test(event.data.message)
```

Comment:
Could we lean on the HTTP protocol here and use status code rather than a message match?